sdk/core/src/dto/shared.dto.ts

```diff
@@ -164,11 +173,11 @@ export class CredentialParams {
 
   @IsDefined()
   @IsArray()
-  context: FreeFormObject[]
+  context: unknown[]
```

Comment:
what problems we were having with freeformobject? and why did we remove it? removing it can cause backwards incompatible changes where we return these types. and the client will need to do type checks / guards where he was previously accessing FreeFormObjects wthout any checks.

Reply:
I've removed `FreeFormObject` because it was not worth it. It increases the validation complexity significantly, is recursive (of arbitrary depth), often does not cover some cases (e.g. some field that is usually a `string[]` was declared as `FreeFormObject` but we didn't have type checking at the moment; after I enabled type checking for that field (by removing some `as any`), it started to fail).

It is exported and then used in services (such as cloud-wallet-api) for tsoa validation, which means that it is not even typescript, that it should be written in tsoa-compliant subset of typescript (so SDK would have to know implicitly about tsoa; and in the future a developer could introduce a benign change to the SDK which would break tsoa compatibility without that developer even suspecting).

In most of the cases, data typed as `FreeFormObject` is subject to some more specific validation later according to its specific schema anyway, so not much sense in validating this data to be `FreeFormObject` beforehand.

And after trying to cover all cases, which would look like this:
```type FreeFormObject = string | FreeFormObject[] | Record<string, FreeFormObject>```
which basically means that it is something compatible with JSON and not containing any bools or numbers.
Are we sure that data typed as FreeFormObject should not contain bools or numbers?
Is it worth it to maintain all that complexity just to have this restriction declared on TS level (even though the specific data objects will still be validated by their specific schemas)?

Reply:
> removing it can cause backwards incompatible changes where we return these types.

This is an input parameter, and clients can supply any array (including an array of former `FreeFormObject`s) to a method expecting `unknown[]`.